File: src/initial_values.cpp
#include <cmath>

#include "initial_values.hpp"

void generate_initial_positions(double positions[][2],int row_size, int num, double r) {

    int k=0;
    int col_size = row_size;
    
    for (int i = 0; i < col_size; ++i)
    {
        for (int j = 0; j < row_size; ++j) {

            if (i % 2 == 0) {
                positions[k][X] = std::sqrt(3)*r/2 * i + r/2;
                positions[k][Y] = r * j + r/2 ;
                ++k;
            }
            else {
                positions[k][X] = std::sqrt(3)*r/2 * i + r/2;
                positions[k][Y] = r * j + r / 2 + r / 2;
                ++k;
            }
        }
    }
}

void generate_initial_velocities(double velocities[][2], int NUM) {
    for (int i = 0; i < NUM; ++i) {
        velocities[i][X] = 1.0;
        velocities[i][Y] = 0.0;
    }
}

File: src/initial_values.py
import numpy as np

def generate_initial_positions(row_size, num, r):
    """
    初期位置を生成します。

    Parameters:
        row_size: 行のサイズ（未使用）
        num: 粒子数（未使用）
        r: 半径（未使用）

    Returns:
        positions: 初期位置を格納した (2, 2) 配列
    """
    positions = np.zeros((2, 2))  # 2 x 2 の配列を生成
    positions[0, 0] = 1.0
    positions[0, 1] = 2.5
    positions[1, 0] = 4.0
    positions[1, 1] = 2.5
    return positions

def generate_initial_velocities(num):
    """
    初期速度を生成します。

    Parameters:
        num: 粒子数（未使用）

    Returns:
        velocities: 初期速度を格納した (2, 2) 配列
    """
    velocities = np.zeros((2, 2))  # 2 x 2 の配列を生成
    velocities[0, 0] = 1.0
    velocities[0, 1] = 0.0
    velocities[1, 0] = 0.0
    velocities[1, 1] = 0.0
    return velocities

